packages/data-service/src/data-service.ts

```diff
@@ -1291,62 +1301,41 @@ export class DataServiceImpl implements DataService {
     if (encryptedFieldsInfo) {
       options.encryptedFields = encryptedFieldsInfo;
     }
-    try {
-      const result = await coll.drop(options);
-      logop(null, result);
-      return result;
-    } catch (error) {
-      logop(error);
-      throw this._translateErrorMessage(error);
-    }
+    return await coll.drop(options);
   }
 
+  @op(mongoLogId(1_001_000_040), ([db], result) => {
+    return { db, ...(result && { result }) };
+  })
   async dropDatabase(name: string): Promise<boolean> {
-    const logop = this._startLogOp(
-      mongoLogId(1_001_000_040),
-      'Running dropDatabase',
-      { db: name }
-    );
     const db = this._database(name, 'CRUD');
-    try {
-      const result = await db.dropDatabase();
-      logop(null, result);
-      return result;
-    } catch (error) {
-      logop(error);
-      throw this._translateErrorMessage(error);
-    }
+    return await db.dropDatabase();
   }
 
+  @op(mongoLogId(1_001_000_180), ([ns, name], result) => {
+    return { ns, name, ...(result && { result }) };
+  })
   async dropIndex(ns: string, name: string): Promise<Document> {
-    const logop = this._startLogOp(
-      mongoLogId(1_001_000_060),
-      'Running dropIndex',
-      { ns, name }
-    );
     const coll = this._collection(ns, 'CRUD');
-    try {
-      const result = await coll.dropIndex(name);
-      logop(null, result);
-      return result;
-    } catch (err) {
-      logop(err);
-      throw this._translateErrorMessage(err);
-    }
+    return await coll.dropIndex(name);
   }
 
+  @op(mongoLogId(1_001_000_041), ([ns, pipeline]) => {
+    return { ns, stages: pipeline.map((stage) => Object.keys(stage)[0]) };
+  })
   aggregateCursor(
     ns: string,
     pipeline: Document[],
     options: AggregateOptions = {}
   ): AggregationCursor {
-    log.info(mongoLogId(1_001_000_041), this._logCtx(), 'Running aggregation', {
-      ns,
-      stages: pipeline.map((stage) => Object.keys(stage)[0]),
-    });
     return this._collection(ns, 'CRUD').aggregate(pipeline, options);
   }
 
+  // @ts-expect-error generic in the method trips up TS here resulting in
+  // Promise<unknown> is not assignable to Promise<Document[]>
+  @op(mongoLogId(1_001_000_181), ([ns, pipeline]) => {
+    return { ns, stages: pipeline.map((stage) => Object.keys(stage)[0]) };
+  })
```

Comment:
That's the only place where decorator types couldn't correctly infer the return value type because of a generic, I feel like it's okay to ignore, but wanted to highlight it nevertheless